axon/act_prjn.go

```diff
@@ -126,8 +127,42 @@ func (sc *SynComParams) ReadOff(cycTot int32) uint32 {
 // ReadIdx returns index for reading existing spikes from the GBuf buffer,
 // based on the layer-based recv neuron index and the
 // ReadOff offset from the CycleTot.
-func (sc *SynComParams) ReadIdx(rnIdx uint32, cycTot int32) uint32 {
-	return rnIdx*sc.DelLen + sc.ReadOff(cycTot)
+func (sc *SynComParams) ReadIdx(rnIdx uint32, cycTot int32, nRecvNeurs uint32) uint32 {
+	// return rnIdx*sc.DelLen + sc.ReadOff(cycTot)
+	return sc.ReadOff(cycTot)*nRecvNeurs + rnIdx // delay is outer, neurs are inner -- should be faster?
+}
+
+// FloatToIntFactor returns the factor used for converting float32
+// to int32 in GBuf encoding.  Because total G is constrained via
+// scaling factors to be around ~1, it is safe to use a factor that
+// uses most of the available bits, leaving enough room to prevent
+// overflow when adding together the different vals.
+// For encoding, bake this into scale factor in SendSpike, and
+// cast the result to int32.
+func (sc *SynComParams) FloatToIntFactor() float32 {
```

Comment:
same difference from the compiler's perspective, right?  and consistent with other versions that do a bit of math.  only needed locally so seems better to have it scoped to this type.  There is no const within a type scope, at least on GPU.

Reply:
Not really, it means relying on some compiler optimization pass that isn't guaranteed to run, always better to not rely on implicit compiler behaviour. Plus it's confusing, since this is literally a constant, and should be specified as such.